packages/graphback-runtime-mongodb/src/OffixMongoDBDataProvider.ts

```diff
@@ -10,17 +10,50 @@ import { MongoDBDataProvider } from './MongoDBDataProvider';
  * https://offix.dev/docs/conflict-server#structure-of-the-conflict-error
  */
 export class OffixMongoDBDataProvider<Type = any, GraphbackContext = any> extends MongoDBDataProvider<Type, GraphbackContext> {
+  protected updatedAtField: string;
+  protected createdAtField: string;
 
   public constructor(baseType: GraphQLObjectType, client: any) {
     super(baseType, client);
+    this.createdAtField = undefined;
+    this.updatedAtField = undefined;
+    Object.keys(baseType.getFields()).forEach((k: string) => {
+      if (baseType.getFields()[k]?.astNode?.directives) {
+        baseType.getFields()[k].astNode.directives.forEach((directive: any) => {
+          if (directive?.name?.value === "createdAt") {
+            if (this.createdAtField === undefined) {
+              this.createdAtField = baseType.getFields()[k]?.name;
+            } else {
+              throw Error("Cannot have more than one field with createdAt directive");
+            }
+          }
+          if (directive?.name?.value === "updatedAt") {
+            if (this.updatedAtField === undefined) {
+              this.updatedAtField = baseType.getFields()[k]?.name;
+            } else {
+              throw Error("Cannot have more than one field with updatedAt directive");
+            }
+          }
+        });
+      }
+    });
   }
 
   public async create(data: any): Promise<Type> {
     if (!data.version) {
       data.version = 1;
     }
+    const o = await super.create(data);
+    let res = o as any;
+    if (this.updatedAtField) {
+      res = await super.update({
```

Comment:
Why is there a second mutation in `create`? Can't this all be done in a single create?

Reply:
Same remark as the one I added here https://github.com/aerogear/graphback/pull/1301#discussion_r430042997

Reply:
This is a hack for setting the updatedAt field of a newly created document to be the same as it's createdAt